resources/api/reservation.py

```diff
@@ -134,29 +134,40 @@ def __init__(self, *args, **kwargs):
         # set supported and required extra fields
         if resource:
             cache = self.context.get("reservation_metadata_set_cache")
-            supported = resource.get_supported_reservation_extra_field_names(
-                cache=cache
-            )
-            required = resource.get_required_reservation_extra_field_names(cache=cache)
-
             # staff events have less requirements
             request_user = self.context["request"].user
             is_staff_event = data.get("staff_event", False)
 
+            # staff events always have the same set of required fields
             if is_staff_event and resource.can_create_staff_event(request_user):
                 required = {"reserver_name", "event_description"}
 
+                # billing fields are also required if own use
+                if data.get("type", None) == Reservation.TYPE_NORMAL:
+                    required |= {
+                        "billing_first_name",
+                        "billing_last_name",
+                        "billing_email_address",
+                    }
+            else:
+                required = resource.get_required_reservation_extra_field_names(
+                    cache=cache
+                )
+
+            for field_name in required:
+                if field_name in self.fields:
```

Comment:
I guess this answer's the question above but wanted to double check that this doesn't make the billing fields required for resources that are free.

Reply:
Yes, that should be the case but it wouldn't hurt to add more tests to make sure that is the case.

Reply:
Thinking more about this, perhaps it should be sufficient to check these fields are present in the required metadata i.e.  

```python

            required = resource.get_required_reservation_extra_field_names(cache=cache)

            if is_staff_event and resource.can_create_staff_event(request_user):
                required_for_staff = {"reserver_name", "event_description"}

                # billing fields are also required if own use
                if data.get("type", None) == Reservation.TYPE_NORMAL:
                    required_for_staff |= {
                        field_name
                        for field_name in (
                            "billing_first_name",
                            "billing_last_name",
                            "billing_email_address",
                        )
                        if field_name in required
                    }
                required = required_for_staff

```

Reply:
Yeah, that should do it